ekf/tests/test_runner/main.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Phoenix-Pilot
+ *
+ * EKF test runner
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+
+#include <hmap.h>
+
+#include "config_file_handler.h"
+#include "result_check.h"
+
+
+#define LOG_FILE "ekf_log.bin"
+#define CHANGE
+
+int ekftests_testSetUp(char *scenarioFile)
+{
+	const size_t changes = 4;
+	const char *fieldsToChange[changes];
+	const size_t scenarioFieldSize = MAX_FILE_TO_CHANGE_STR_LEN + 1;
+	char scenarioField[scenarioFieldSize];
+
+	if (strlen(scenarioFile) > MAX_VALUE_LEN) {
+		fprintf(stderr, "Path to scenario file is too long\n");
+		return -1;
+	}
+
+	snprintf(scenarioField, scenarioFieldSize, "DATA_SOURCE/file=%s", scenarioFile);
+
+	fieldsToChange[0] = "DATA_SOURCE/source=LOGS";
+	fieldsToChange[1] = "LOGGING/log=ALL";
+	fieldsToChange[2] = "LOGGING/mode=STRICT";
+	fieldsToChange[3] = scenarioField;
+
+	if (ekftests_configPrepare(fieldsToChange, changes) != 0) {
+		return -1;
+	}
+
+	return 0;
+}
+
+
+inline int ekftests_testTearDown(void)
```

Comment:
I think `inline` may be dropped. Lets put this optimalization step on compiler if it wishes so. However:
```suggestion
static int ekftests_testTearDown(void)
```